modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/parsers/AbstractGetFeatureResponseParserFactory.java

```diff
@@ -102,73 +106,114 @@ public boolean canProcess(final WFSRequest request, final String contentType) {
+        // header to plain "text/xml" instead of "text/xml;subtype=gml/3.1.1". So we'll do a bit
+        // of heuristics to find out what it actually returned
+        final int buffSize;
+        if (LOGGER.isLoggable(Level.FINER)) {
+            buffSize = 4096;
         } else {
-            // We can't rely on the server returning the correct output format. Some, for example
-            // CubeWerx, upon a successful GetFeature request, set the response's content-type
-            // header to plain "text/xml" instead of "text/xml;subtype=gml/3.1.1". So we'll do a bit
-            // of heuristics to find out what it actually returned
-            final int buffSize;
-            if (LOGGER.isLoggable(Level.FINER)) {
-                buffSize = 4096;
-            } else {
-                buffSize = 512;
-            }
-            PushbackInputStream pushbackIn = new PushbackInputStream(response.getResponseStream(),
-                    buffSize);
-            byte[] buff = new byte[buffSize];
-            int readCount = 0;
-            int r;
-            while ((r = pushbackIn.read(buff, readCount, buffSize - readCount)) != -1) {
-                readCount += r;
-                if (readCount == buffSize) {
-                    break;
-                }
+            buffSize = 512;
+        }
+        PushbackInputStream pushbackIn = new PushbackInputStream(response.getResponseStream(), buffSize);
+        byte[] buff = new byte[buffSize];
+        int readCount = 0;
+        int r;
+        while ((r = pushbackIn.read(buff, readCount, buffSize - readCount)) != -1) {
+            readCount += r;
+            if (readCount == buffSize) {
+                break;
             }
+        }
 
-            String charset = response.getResponseHeader("Charset");
-            try {
-                Charset.forName(charset);
-            } catch (Exception e) {
-                charset = "UTF-8";
-            }
+        String charset = response.getResponseHeader("Charset");
+        try {
+            Charset.forName(charset);
+        } catch (Exception e) {
+            charset = "UTF-8";
+        }
 
-            BufferedReader reader = new BufferedReader(new InputStreamReader(
-                    new ByteArrayInputStream(buff), charset));
-            StringBuilder head = new StringBuilder();
+        StringBuilder head = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(
+                new ByteArrayInputStream(buff), charset))) {
             String line;
             while ((line = reader.readLine()) != null) {
                 head.append(line).append('\n');
             }
-            if(LOGGER.isLoggable(Level.FINER)){
-                System.err.println("Response head:");
-                System.err.println(head);
+        }
+        if (LOGGER.isLoggable(Level.FINER)) {
+            System.err.println("Response head:");
+            System.err.println(head);
```

Comment:
These ```System.err``` statements flood the logs and cause Travis CI to fail. Something changed in the logic? I am going to replace this block with:
```
if (LOGGER.isLoggable(Level.FINER)) {
    LOGGER.finer("Response head:\n" + head);
}
```